app/controllers/proposals_controller.rb

```diff
@@ -11,6 +11,7 @@ class ProposalsController < ApplicationController
   rescue_from Pundit::NotAuthorizedError, with: :auth_errors
 
   def show
+    fail "User #{@current_user} has no client-slug" if @current_user.client_slug.blank?
```

Comment:
What's the best way to check this? Somewhere in the authorization code? It's a little complex.

I added this as a quick diagnostic to eliminate nil errors.